### PR TITLE
Convert HeaderBlock into RSC

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Convert `HeaderBlock` (our site's nav header) into server component
- Split child blocks into their own client components
- Removed few uses of Emotion we had in the header

Almost every change is an automated refactoring, I'll comment places with other changes

## Justify why they are needed

Bring Header closer to recommended code organization before we apply new design to it

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
* **#4387** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4387?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/blocks/HeaderBlock/ProductNavContainerBlock.css.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { style } from '@vanilla-extract/css'
+import { tokens } from 'ui/src/theme'
+
+export const mobileWrapper = style({
```
Comment (alebedev) [inline apps/store/src/blocks/HeaderBlock/ProductNavContainerBlock.css.ts:4]: emotion -> vanilla

File: apps/store/src/blocks/HeaderBlock/ProductNavContainerBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                </ProductNavigationLink>
+              </NavigationMenuPrimitive.Item>
+            ))}
+          </NavigationMenuPrimitive.List>
+        </NavigationMenuPrimitive.Sub>
+        <ButtonNextLink
+          href={PageLink.store({ locale })}
+          variant="secondary"
+          size="medium"
+          fullWidth={true}
```
Comment (alebedev) [inline apps/store/src/blocks/HeaderBlock/ProductNavContainerBlock.tsx:82]: using supported prop instead of custom emotion style here

**Review by guilhermespopolin — APPROVED** [review]
Well done :)

**Comment by alebedev** [comment]
### Merge activity

* **Jun 20, 4:35 AM EDT**: @alebedev merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4387).

**Pull request merged**